pkg/cfaws/assumer_aws_sso.go

```diff
@@ -154,12 +154,26 @@ func SSODeviceCodeFlow(ctx context.Context, cfg aws.Config, rootProfile *CFShare
 	// trigger OIDC login. open browser to login. close tab once login is done. press enter to continue
 	url := aws.ToString(deviceAuth.VerificationUriComplete)
 	fmt.Fprintf(color.Error, "If browser is not opened automatically, please open link:\n%v\n", url)
-	err = browser.OpenURL(url)
+
+	//check if sso browser path is set
+
+	config, err := grantCfg.Load()
 	if err != nil {
-		// fail silently
-		debug.Fprintf(debug.VerbosityDebug, color.Error, err.Error())
+		return nil, err
+	}
+	if config.CustomSSOBrowserPath != "" {
+		err = browsers.LaunchBrowserSession(url)
```

Comment:
From what I can tell, launch browser session also includes logic to prompt to launch the url manually, though we have been adding these messages where we use the browser. 

In this section, we already print a message above to prompt the user to open the browser manually.

It also looks like you have left the default browser behaviour commented out. 

I think it should be in an else statement.


I propose that the LaunchBrowserSession function is renamed to OpenUrlWithCustomBrowser and it should behave the same way as browser.OpenUrl()